pages/api/pearl-daa.js

```diff
@@ -0,0 +1,216 @@
+    return acc;
+  }, {});
+};
+
+const getPerformanceByAgentId = async () => {
+  // Only use valory_trader agent IDs
+  const traderAgentIds = PREDICT_AGENT_CLASSIFICATION.valory_trader.map((n) =>
+    Number(n),
+  );
+
+  // Fetch 8 days of data (7 + 1 buffer day for completeness)
+  const timestamp_lt = getMidnightUtcTimestampDaysAgo(0);
+  const timestamp_gt = getMidnightUtcTimestampDaysAgo(8);
+
+  try {
+    const performance = await REGISTRY_GRAPH_CLIENTS.gnosis.request(
+      predictDailyAgentPerformancesQuery,
+      {
+        agentId_in: traderAgentIds,
+        dayTimestamp_gte: timestamp_gt,
+        dayTimestamp_lte: timestamp_lt,
+      },
+    );
+
+    return performance?.dailyAgentPerformances ?? [];
+  } catch (error) {
+    console.error('GraphQL query error:', error);
+    return [];
+  }
+};
+
+const groupServicesByDays = (activeServices, pearlServiceIds) => {
+  const servicesByDays = {};
+
+  for (const service of activeServices) {
+    const date = new Date(Number(service.dayTimestamp) * 1000)
+      .toISOString()
+      .slice(0, 10);
+
+    if (!servicesByDays[date]) {
+      servicesByDays[date] = new Set();
+    }
+
+    for (const multisig of service.multisigs) {
+      const serviceId = Number(multisig.multisig.serviceId);
+      if (pearlServiceIds.has(serviceId)) {
+        servicesByDays[date].add(multisig.multisig.id);
+      }
+    }
+  }
+
+  return servicesByDays;
+};
+
+const getOptimusDailyCounts = async () => {
+  const timestamp_lt = getMidnightUtcTimestampDaysAgo(0);
+  const timestamp_gt = getMidnightUtcTimestampDaysAgo(8);
+
+  try {
+    const result = await REGISTRY_GRAPH_CLIENTS.optimism.request(
+      dailyBabydegenPerformancesQuery,
+      {
+        timestamp_gt,
+        timestamp_lt,
+      },
+    );
+
+    const performances = result.dailyAgentPerformances ?? [];
+
+    const dailyCounts = {};
+
+    performances.forEach((perf) => {
+      // Extract timestamp from id: "day-{timestamp}-agent-40"
+      const timestampMatch = perf.id.match(/day-(\d+)-agent-40/);
+      if (!timestampMatch) {
+        console.error('Invalid id format:', perf.id);
+        return;
+      }
+      const timestamp = Number(timestampMatch[1]);
+      const date = new Date(timestamp * 1000).toISOString().slice(0, 10);
+      dailyCounts[date] = Number(perf.activeMultisigCount);
+    });
+    return dailyCounts;
+  } catch (error) {
+    console.error('Error fetching Optimism DAAs:', error);
+    return {};
+  }
+};
+
+const calculate7DayTrailingAverage = (dailyCounts) => {
+  if (dailyCounts.length < 7) {
+    return 0;
+  }
+
+  const sum = dailyCounts.reduce((acc, count) => acc + count, 0);
+  return sum / 7;
+};
+
+const getCombinedPearlDAAs = async () => {
+  const [pearlDailyCounts, optimusDailyCounts] = await Promise.all([
+    getPearlDailyCounts(),
+    getOptimusDailyCounts(),
+  ]);
+
+  const allDates = new Set([
+    ...Object.keys(pearlDailyCounts),
+    ...Object.keys(optimusDailyCounts),
+  ]);
```

Comment:
please use `Promise.allSettled` for the above, if one of the calls fails, we won't see any data